### PR TITLE
[Ruby 2.5] Add spec for Thread#fetch

### PR DESCRIPTION
**Pull request by Atul9**
#565

### CODE REVIEW

File: core/thread/fetch_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ruby_version_is '2.5' do
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: Can you require the `spec_helper` here like in other files?

File: core/thread/fetch_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  describe 'Thread#fetch' do
+    it "returns the value of the fibre-local varible if value has been assigned" do
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: fiber-local

File: core/thread/fetch_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      th.join
+      th.fetch(:cat, true).should == 'meow'
+    end
+
+    it "returs the default value if fibre-local variable hasn't been assigned" do
+      th = Thread.new {}
+      th.join
+      th.fetch(:cat, true).should == true
+    end
+  end
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: Can you add specs for cases where only 1 argument is passed to Thread#fetch?

File: core/thread/fetch_spec.rb (hunk 1 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: Do we need it? I ran the specs locally it passed. Hence I didn't include it.

Comment (eregon) [reply core/thread/fetch_spec.rb]: It's notably useful to run the spec directly like `$ ruby core/thread/fetch_spec.rb`.
There is also some setup related to `Thread.report_on_exception` which should be done before any spec.

File: core/thread/fetch_spec.rb (hunk 3 above)
Comment (eregon) [reply core/thread/fetch_spec.rb]: No, I think KeyError is intended and consistent with `Hash#fetch`.

File: core/thread/fetch_spec.rb (hunk 1 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: done. I have updated the PR.

File: core/thread/fetch_spec.rb (hunk 2 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: done. I have updated the PR.

File: core/thread/fetch_spec.rb (hunk 3 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: okay. I have updated the PR.

File: core/thread/fetch_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      th.fetch(:cat, true).should == 'meow'
+    end
+
+    it "returns the default value if fiber-local variable hasn't been assigned" do
+      th = Thread.new {}
+      th.join
+      th.fetch(:cat, true).should == true
+    end
+
+    it 'raises a KeyError when only one argument is passed' do
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: `raises a KeyError when the Thread does not a fiber-local variable of same name`

File: core/thread/fetch_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      th = Thread.new {}
+      th.join
+      -> { th.fetch(:cat) }.should raise_error(KeyError)
+    end
+
+    it 'raises an ArgumentError when not passed one or two arguments' do
+      th = Thread.new {}
+      th.join
+      -> { th.fetch() }.should raise_error(ArgumentError)
+      -> { th.fetch(1, 2, 3) }.should raise_error(ArgumentError)
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: Could you use `Thread.current` for these examples?

File: core/thread/fetch_spec.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      th = Thread.new {}
+      th.join
+      th.fetch(:cat, true).should == true
+    end
+
+    it 'raises a KeyError when only one argument is passed' do
+      th = Thread.new {}
+      th.join
+      -> { th.fetch(:cat) }.should raise_error(KeyError)
+    end
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: There should be a complementary spec when `:cat` is set, so then `th.fetch(:cat)` returns the value.

File: core/thread/fetch_spec.rb (hunk 4 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: done. I have updated the PR.

File: core/thread/fetch_spec.rb (hunk 6 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: okay, I have added new spec & updated the PR.

File: core/thread/fetch_spec.rb (hunk 5 above)
Comment (Atul9) [reply core/thread/fetch_spec.rb]: yes, I can use Thread.current. I have updated the PR.

File: core/thread/fetch_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      th.fetch(:cat, true).should == true
+    end
+
+    it 'raises a KeyError when the Thread does not have a fiber-local variable of the same name' do
+      th = Thread.new {}
+      th.join
+      -> { th.fetch(:cat) }.should raise_error(KeyError)
+    end
+
+    it 'returns the value of the fiber-local varible if value has been assigned' do
```
Comment (eregon) [inline core/thread/fetch_spec.rb]: This spec has the same description as the first one.
This is problematic when tagging (excluding) a spec.
I think an elegant solution here would be to wrap sepcs 1-2 in `describe "with 2 arguments" do` and 3-4 in `describe "with 1 argument" do`

Comment (eregon) [reply core/thread/fetch_spec.rb]: And `varible` => `variable` :)

Comment (Atul9) [reply core/thread/fetch_spec.rb]: done. I have updated the PR.

**Pull request merged**